app/packages/state/src/recoil/schema.ts

```diff
@@ -220,14 +212,17 @@ export const fieldPaths = selectorFamily<
         throw new Error("path and space provided");
       }
 
-      const sampleLabels = Object.keys(
-        get(fieldSchema({ space: State.SPACE.SAMPLE, flat: true }))
-      )
+      const sampleFields = get(
+        fieldSchema({ space: State.SPACE.SAMPLE, flat: true })
+      );
+      const frameFields = get(
+        fieldSchema({ space: State.SPACE.FRAME, flat: true })
+      );
+
+      const sampleLabels = Object.keys(sampleFields)
         .filter((l) => !l.startsWith("_"))
         .sort();
-      const frameLabels = Object.keys(
-        get(fieldSchema({ space: State.SPACE.FRAME, flat: true }))
-      )
+      const frameLabels = Object.keys(frameFields)
```

Comment:
Nice, This is easier to follow.
We should add a comment about do not remove 'flat:true'